types/node/globals.d.ts

```diff
@@ -47,15 +47,15 @@ declare var console: Console;
 declare var __filename: string;
 declare var __dirname: string;
 
-declare function setTimeout(callback: (...args: any[]) => void, ms?: number, ...args: any[]): NodeJS.Timeout;
+declare function setTimeout<T extends any[]>(callback: (...args: T) => void, ms?: number, ...args: T): NodeJS.Timeout;
 declare namespace setTimeout {
     function __promisify__(ms: number): Promise<void>;
-    function __promisify__<T>(ms: number, value: T): Promise<T>;
+    function __promisify__<T>(ms: number, value: T, options?: TimerOptions): Promise<T>;
```

Comment:
In order to use this you should probably move this into the timer module using the `global` directive.